include/linker/common-rom.ld

```diff
@@ -118,36 +108,16 @@
 	} GROUP_LINK_IN(ROMABLE_REGION)
 #endif
 
-	SECTION_DATA_PROLOGUE(_bt_channels_area,,SUBALIGN(4))
-	{
-		_bt_l2cap_fixed_chan_list_start = .;
-		KEEP(*(SORT_BY_NAME("._bt_l2cap_fixed_chan.static.*")))
-		_bt_l2cap_fixed_chan_list_end = .;
-	} GROUP_LINK_IN(ROMABLE_REGION)
+	Z_ITERABLE_SECTION_ROM(bt_l2cap_fixed_chan, 4)
 
 #if defined(CONFIG_BT_BREDR)
```

Comment:
For all these, it seems like the #ifdef is needless.  There's no alignment being done or anything else that will be visible outside the link (beyond the existence of the start/end symbols).  Seems like we should be able to include these under all conditions.

I guess the argument against would be "what if there was stuff in the "_bt_l2cap_br_fixed_chan" section when CONFIG_BT_BREDR is unset?".  In which my answer would be: isn't that a bug already?

Reply:
Looking at this, yes I agree there are a bunch of conditional ifdefs in these files (not just CONFIG_BT_BREDR, there are many more) that could be removed with no ill effects. Should be in a separate patch though.